libs/providers/openstack.py

```
@@ -131,13 +134,43 @@ def get_flavor_obj(self, vm_name: str) -> Any:
         instance_id = self.get_instance_id_by_name(name_filter=vm_name)
-        # Get the volume attachments associated with the instance
-        volume_attachments = self.api.compute.volume_attachments(server=self.api.compute.get_server(instance_id))
+        volume_attachments = list(self.api.compute.volume_attachments(server=self.api.compute.get_server(instance_id)))
+
         for attachment in volume_attachments:
             volume = self.api.block_storage.get_volume(attachment["volumeId"])
-            return volume.volume_image_metadata
+            if volume.is_bootable:
+                return volume.volume_image_metadata
+
+        LOGGER.error(f"No boot volume found for VM '{vm_name}'. Checked {len(volume_attachments)} attachments.")
```

Ruff (0.15.1)

[warning] Logging statement uses f-string

(G004)